devito/types/tensor.py

```diff
@@ -342,8 +346,11 @@ def grad(self, shift=None):
         Gradient of the VectorFunction, creates the gradient TensorFunction.
         """
         func = tens_func(self, self)
```

Comment:
self, self ? same arg?

Reply:
You're right. The same happens on line 247. Also with `vec_func` on lines 222, 237, 322, 362. The only one which uses different argument is at 358. We could simplify this with:
```python 
def tens_func(*funcs):
    return (TensorTimeFunction if any([getattr(f, 'is_TimeDependent', False)
            for f in funcs]) else TensorFunction)
```
and the same with `vec_func`